includes/Commerce/Orders.php

```diff
@@ -82,10 +82,86 @@ public function create_local_order( Order $remote_order ) {
 	 * @param Order $remote_order Orders API order object
 	 * @param \WC_Order $local_order local order object
 	 * @return \WC_Order
+	 * @throws SV_WC_Plugin_Exception|\WC_Data_Exception
 	 */
 	public function update_local_order( Order $remote_order, \WC_Order $local_order ) {
 
-		// TODO: implement
+		// add/update items
+		foreach ( $remote_order->get_items() as $item ) {
+
+			$wc_product_id = $item['retailer_id'];
+
+			$matching_wc_order_item = false;
+
+			// check if the local order already has this item
+			foreach ( $local_order->get_items() as $wc_order_item ) {
+
+				if ( $wc_order_item instanceof \WC_Order_Item_Product && $wc_product_id === (string) $wc_order_item->get_product_id() ) {
+					$matching_wc_order_item = $wc_order_item;
+					break;
+				}
+			}
+
+			if ( empty( $matching_wc_order_item ) ) {
+
+				$wc_product = wc_get_product( $wc_product_id );
+
+				if ( ! $wc_product instanceof \WC_Product ) {
+					throw new SV_WC_Plugin_Exception( "Product with WC ID $wc_product_id not found" );
+				}
+
+				$matching_wc_order_item = new \WC_Order_Item_Product();
+				$local_order->add_item( $matching_wc_order_item );
+			}
+
+			$matching_wc_order_item->set_product_id( $wc_product_id );
+			$matching_wc_order_item->set_quantity( $item['quantity'] );
+			$matching_wc_order_item->set_subtotal( $item['quantity'] * $item['price_per_unit']['amount'] );
+			// TODO: should we use the estimated_tax or the captured_tax on the line below?
```

Comment:
I think we'll be using `estimated_tax` - captured tax represents the tax after the order/item has been shipped, but we don't fulfill orders at the line-item level, only on a full order status change of Completed

Reply:
```suggestion
			// we use the estimated_tax because the captured_tax represents the tax after the order/item has been shipped and we don't fulfill order at the line-item level
```